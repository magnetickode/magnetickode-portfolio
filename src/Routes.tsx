import React from "react";
import { Switch, Route, Link } from "react-router-dom";

import Home from "./pages/home";

const Routes: React.FC = () => (
  <Switch>
    <Route exact path="/" component={Home} />
  </Switch>
);

export default Routes;
