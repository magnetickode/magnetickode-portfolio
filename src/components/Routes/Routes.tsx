import React from "react";
import { Switch, Route } from "react-router-dom";

import Home from "../../pages/home";

const Routes: React.FC = () => (
  <Switch data-test="RoutesComponent">
    <Route exact path="/" component={Home} />
  </Switch>
);

export default Routes;
