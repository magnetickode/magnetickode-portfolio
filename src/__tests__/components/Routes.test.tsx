import React from "react";
import { shallow, ShallowWrapper } from "enzyme";

import Routes from "../../components/Routes";

let wrapper: ShallowWrapper;

beforeEach(() => {
  wrapper = shallow(<Routes />);
});

it("matches snapshot", () => {
  expect(wrapper).toMatchSnapshot();
});

it("renders without error", () => {
  const RoutesComponent = wrapper.find("[data-test='RoutesComponent']");
  expect(RoutesComponent.length).toBe(1);
});
