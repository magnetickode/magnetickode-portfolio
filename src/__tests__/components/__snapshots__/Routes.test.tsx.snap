// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Switch
  data-test="RoutesComponent"
>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
</Switch>
`;
